### DISCUSSION TITLE
I need help with PRs, issues and Slack.

### DISCUSSION THREAD

**Discussion by Sean-Der** [pion/webrtc#1836] [General]
Recently my personal life has gotten much busier (work, moving and a baby). I haven't been able to keep up with Pion's needs lately so would really like some help. Quite a few PRs have become blocked on me. We are loosing good contributors because no one is able to merge their code.

## Can someone find build tooling to surface issues/PRs
It would be useful to know how many issues/PRs we have across all the repos. Maybe something already exists or is built into GitHub?

## Who else can review/mentor contributors?
Anyone can approve/merge a PR. I think people are worried that code needs to be perfect, it is ok to merge something to master that is WIP. I am happy with the checks we have, we just need to make sure the following is true.

* Don't break the API
* When possible follow the Javascript API
* When adding new features make sure we can modify/extend them in the future

@at-wat @enobufs @backkem @daenney you have all been successful at getting involved in Pion independently. Any advice on how we can encourage others?

## We need to finish out some high important issues.

* [Interceptors don't allow re-use](https://github.com/pion/webrtc/issues/1749)
* [Detached DataChannels OnOpen Race](https://github.com/pion/webrtc/issues/1063)
* [Transport Wide Congestion Control](https://github.com/pion/interceptor/pull/40)
* More?

## What is next?
What features would bring the most value to users next? I don't have a clear vision yet what people need the most. These are some of the things I hear the most.

* Pure Go Congestion Controller
* TinyGo support
* Automated Performance Testing
* More webrtc-stats

**Comment by Sean-Der** [comment]
From @Antonito on Slack

```
I only got back on WebRTC recently as you know, but here are some points I found confusing / unclear (and I guess others too):
Media Engine default codecs are great, but they don’t seem to support the codec libwebrtc uses on iOS – and I don’t think they can be added since the Payload Types conflict? (ex: iOS uses H264 with PayloadType 96 and 98, Pion registers VP8 and VP9 for those values). Maybe Pion could have a set of ‘codec lists’, instead of a single default one? (default for chrome, for firefox, for android native, ..etc)
It seems some tests randomly fail. Not a big deal, but I faced this with a single contribution – I guess it can bother bigger/more regular contributors. 

* Regarding the `SetICEUDPMux` and `SetICETCPMux` options, if enabled, does it means a Pion instance can be deployed in a container with only two ports open, and ingress/egress allowing traffic from those two ports only? Are there any drawbacks to this? (performance wise maybe?) That was my original understanding but I’m not sure I got it right (maybe the documentation could be just a little bit more explicit?), but it seems it doesn’t work (most likely be an issue on my end though)

Overall I’m really happy with /v3, I re-coded my SFU from scratch – it took me ~5 days to have everything working, tested & benchmarked. I did a similar thing last summer, with v2/v3-beta, it took me ~1.5 month to achieve pretty much the same result!
In my previous version, I had some low level tests with tracks on which I wrote both `media.Sample` and `rtp.Packet`. It seems it’s not possible anymore with v3.

Totally agree with performance testing + it would give a baseline for expected performance (ex: on my SFU a peer connects in xxx seconds: it’s expected performance / it’s too slow I made a mistake somewhere)
also +1 for stats, I think it’s really missing right now, it would be great to be able to collect info about rtt / bytes inbound / outbound / packet loss, ..etc
```

**Comment by daenney** [comment]
I mostly got here because of my own pedantry; refusing to use CGO bindings to OpenSSL for some DTLS stuff. I got greeted by some welcoming and enthusiastic folks so I decided to try and contribute. I've stuck around mostly by accident :sweat_smile: and because I do want to see this project continue. I personally don't have any investment in the WebRTC side of things though.

I recognise the situation you're in. There's a bunch of stuff in dtls I'd like to fix and rework, but I don't have much time right now. The free time I do have I'm now more actively putting towards doing more leisurely things, because it started to feel like I'd gone without a break for weeks on end at some point.

I'm at the point where I'm wondering if we can get funding for Pion somehow, so we can have folks work on the WebRTC stuff because it's now actually their work.

**Reply by backkem** [reply]
My situation is quite similar. I saw people were failing to do WebRTC data channels (outside of the browser) and I wanted to fill that void. Today, I'm also lacking time & energy to contribute actively.

I've also wondered about funding. I find it a tricky subject since you somehow need to allocate it towards a comprehensive vision. I'd be willing to chip in but at the same time altruistically funding doesn't make for a reliable source of someone's income.
